### ISSUE TITLE
Notifications are never sent with the email_from defined in the template

### ISSUE THREAD

**Issue by ruifranco** [odoo/odoo#21065]
Impacted versions: 
v11

Steps to reproduce: 
Change the notification email template and add an email in the "from" field.

Current behavior:
The user email is always assumed as the sender of the mail message

Expected behavior:
If one defines a sender, this should become the message's sender address

Video/Screenshot link (optional):

**Comment by rruebner** [comment]
I can confirm the issue. We "fixed" it in one of our projects in version 8.
@Yenthe666 Can you confirm this as bug or is it designed to work in this way? If it is a bug we can provide a solution.

**Comment by ruifranco** [comment]
Say, rruebner, perhaps you could share your fix anyway. ;)

**Comment by Yenthe666** [comment]
I believe this was done by intention but I'll let somebody from Odoo itself answer this officially. 😉 
@mart-e can you do the honours?

**Comment by mart-e** [comment]
I assume it is indeed posted as the admin as it is usually done for automated actions.
But what which notification email are you talking about? Sent in which context?

**Comment by ruifranco** [comment]
@mart-e , this happens with the emails sent using chatter (no template chosen).

A company I work for gets spam alerts because, rather than using the "from" address they've defined in the default template, odoo uses the user's email as "from". They've managed to overcome this with some server-side email configurations but all would have worked in the first place should the chatter emails be sent using the address in the default template.

**Comment by ruifranco** [comment]
@mart-e , yes: if the chatter messages follow a default template, than, they should be sent using the "from" email address in the template rather than the user's (that's the way email templates work).

**Comment by ruifranco** [comment]
This problem continesd in v12.

I've changed the /mail/wizard/mail_compose_message.py/get_mail_values method and added 

try:
    mail_values['email_from'] = self.env['mail.template']._render_template(self.template_id.email_from, self.model, res_id)
except Exception as e:
    _logger.info(e)
    pass

just before "# mass mailing: rendering override wizard static values"

**Comment by tde-banana-odoo** [comment]
Normally email_from is correctly computed unless you go through some specific use cases. What is precisely your issue ? Could you give a way to reproduce ?

**Comment by ruifranco** [comment]
Edit the "Invoice: Send by email" email template and change the default "email_from".

Now, go to an invoice and send it by email, using the above mentioned template.

The email that shows in the Settings/Technical/Email/Emails will have the user's email as sender and not the one set in the email template.

I keep getting this in v12, EE and CE (different setups, servers and OS). 
It also happens with a new database with demo data.

Please check the following video:
https://youtu.be/XbCAag_Kds0

**Comment by C3POdoo** [comment]
Dear @ruifranco,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
